File: template/app/src/main/java/com/xmartlabs/template/ui/onboarding/OnboardingView.java
package com.xmartlabs.template.ui.onboarding;

import android.content.Intent;
import android.support.annotation.NonNull;

import com.xmartlabs.template.ui.common.TemplateView;

public interface OnboardingView extends TemplateView {
  OnboardingPageAdapter createPageAdapter();
  void setupView(@NonNull OnboardingPageAdapter pageAdapter);
  void setSkipButtonVisibility(boolean visible);
  void moveToPage(int page);
  void startActivity(Intent intent);
  void handleNextButtonVisibility();
  void goToLoginActivity();
}

File: template/app/src/main/java/com/xmartlabs/template/ui/onboarding/OnboardingView.kt
package com.xmartlabs.template.ui.onboarding

import android.content.Intent

import com.xmartlabs.template.ui.common.TemplateView

interface OnboardingView : TemplateView {
  fun createPageAdapter(): OnboardingPageAdapter
  fun setup(pageAdapter: OnboardingPageAdapter)
  fun setSkipButtonVisibility(visible: Boolean)
  fun moveToPage(page: Int)
  fun startActivity(intent: Intent)
  fun handleNextButtonVisibility()
  fun goToLoginActivity()
}
